dbms/src/Flash/Coprocessor/CHBlockChunkCodecV1.cpp

```diff
@@ -96,12 +97,20 @@ static inline void decodeColumnsByBlock(ReadBuffer & istr, Block & res, size_t r
         return;
 
     auto && mutable_columns = res.mutateColumns();
-    for (auto && column : mutable_columns)
-    {
-        if (reserve_size > 0)
-            column->reserve(std::max(rows_to_read, reserve_size));
-        else
-            column->reserve(rows_to_read + column->size());
+    const auto & name_and_type_list = res.getColumnsWithTypeAndName();
+    size_t column_size = mutable_columns.size();
+    for (size_t i = 0; i < column_size; ++i)
+    {
+        auto && column = mutable_columns[i];
+        /// For non-fixed size type, reserve function might cause too much memory usage, i.e. string type column reserves 64 bytes
```

Comment:
Maybe we can change the codec format and add field about data size for the columns with non-fixed size type. But such way need to change codec & mpp version.

Reply:
Yes, nice suggestion, I plan to introduce some mechanism to "guess" the every data size, since we need to reserve the column size before knowning how many blocks will be squashed.